File: src/node/magma/app/Platform.ts
export class Platform {
	static TypeScript: Platform = new Platform("node", "ts");
	static Magma: Platform = new Platform("magma", "mgs");
	static Windows: Platform = new Platform("windows", "h", "c");
	root: string;
	extension: string[];
	constructor (root: string, ...extensions: string[]) {
		this.root = root;
		this.extension = extensions;
	}
}

File: src/java/magma/app/Platform.java
package magma.app;

public enum Platform {
    TypeScript("node", "ts"),
    Magma("magma", "mgs"),
    Windows("windows", "h", "c");

    final String root;
    final String[] extension;

    Platform(final String root, final String... extensions) {
        this.root = root;
        this.extension = extensions;
    }
}
